### ISSUE TITLE
example setup global markdownstyle -- not working in Debian

### ISSUE THREAD

**Issue by nbehrnd** [markdownlint/markdownlint#535]
## Description

Attempts to replicate the example how to use a global style (as provided by @AJRepo by December 1, 2025) in Linux Debian 14/forky (branch testing) fail.

Question to @AJRepo:  Did you use a different operating system to prepare the PR, i.e. is this perhaps an issue _specific_ to Debian?

### Environment

**MDL Version** 0.15.0 as package 0.15.0-1 for Debian 14/forky ([tracker](https://tracker.debian.org/pkg/ruby-mdl)) available since today.

### Expected Behavior

With a .md file intentionally containing a line much longer than the default threshold, the linter is expected to report this issue.

### Actual Behavior

Though the style file was copy-pasted in the home directory, it is not accessible to markdownlint.  Neither a `source ~/.bashrc`, nor a reboot of the Linux instance improved the situation.  Still specific to Debian 14/forky, the issue is absent if the syntax in path `markdowlint/example/.mdlrc_example` (commit [e4efbcb](https://github.com/markdownlint/markdownlint/commit/e4efbcbc3bad36b50a8d6071ae8ba8dcfb0a8c37), Jul 30, 2020) is used, i.e.

```
style "#{File.dirname(__FILE__)}/.exclude_example.rb"
```

## Replication Case

This section uses the syntax as proposed by @AJRepo:

```bash
@debian:~/Desktop$ cat long_line.md 
# header
Here is a short line
and an other
- list of 
- bullets
- mdl should see this
Again a short line.
And here is the very line, intentionally set to be overly long to issue a warning based on rule MD013.
@debian:~/Desktop$ 
@debian:~/Desktop$ cat ~/.mdlrc 
#The custom style file is referenced by .mdlrc as below
style ".exclude_example.rb"
#style "#{File.dirname(__FILE__)}/.exclude_example.rb"
@debian:~/Desktop$ 
@debian:~/Desktop$ cat ~/.exclude_example.rb 
# This is an example style file.
#
# This file defines the rules to include/exclude and/or parameters
# of those rules.
# This must be included from your .mdlrc using the `style` paramter.

# This is a sample style to exclude very long lines

# First, include all rules:
all
# Then, exclude MD013 (long lines), with:
exclude_rule 'MD013'

# The above excludes all long lines. If, instead, you want to only
# exclude code blocks with long lines, use the following:
# rule 'MD013', :ignore_code_blocks => true
@debian:~/Desktop$ 
@debian:~/Desktop$ mdl -V
0.15.0
@debian:~/Desktop$ 
@debian:~/Desktop$ 
@debian:~/Desktop$ mdl long_line.md 
Style '.exclude_example.rb' does not exist.
```

Now with the syntax described by @Antares1980 which works in Debian the expected way:

```bash
@debian:~/Desktop$ cat ~/.mdlrc
#The custom style file is referenced by .mdlrc as below
# style ".exclude_example.rb"
style "#{File.dirname(__FILE__)}/.exclude_example.rb"
@debian:~/Desktop$ 
@debian:~/Desktop$ mdl long_line.md 
long_line.md:4: MD009 Trailing spaces
long_line.md:1: MD022 Headers should be surrounded by blank lines
long_line.md:4: MD032 Lists should be surrounded by blank lines
long_line.md:6: MD032 Lists should be surrounded by blank lines
```

If it turns out to be an issue specific to Debian, I'm willing to prepare a patch to the documentation in Debian's man page because the global setup complements using markdownlint's rules per flag/on the CLI and the one per local style file.

**Comment by AJRepo** [comment]
The work I did was in Debian, so, no, this isn't a Debian specific issue. 

You have put your `.mdlrc` and  `.exclude_example.rb`  in `~` yet  your working directory and  `long_line.md` is in `~/Desktop` which is a different directory from `~`.  

See  https://github.com/markdownlint/markdownlint/blob/main/docs/creating_rules.md  and https://github.com/markdownlint/markdownlint/blob/main/docs/configuration.md 

The example I created was referenced to a local directory override, overriding a global setup. The use-case is for a developer who works on many different projects across many different organizations where said projects are  organized by directories (e.g. git clone) and said projects are  tied in to different automated CI tests.

**Comment by nbehrnd** [comment]
@AJRepo Indeed, the error was on my side to recognize the global use case by @Antares1980 is separate from the one you highlight.  And the _per project_ approach with a `.mdlrc` indicating a style file as an alternative to a pattern like

```bash
mdl -r mystyle.rb example.md
```

I was used to.  Thank you for the clarification.
